lib/src/logging/logging.cc

```diff
@@ -24,13 +51,16 @@ namespace facter { namespace logging {
 
     void setup_logging(ostream& os)
     {
-        // Initialize boost filesystem's locale to a UTF-8 default.
-        // Logging gets setup the same way via the default 2nd argument.
-#if !defined(__sun) || !defined(__GNUC__)
-        // Locale support in GCC on Solaris is busted, so skip it.
-        boost::filesystem::path::imbue(leatherman::locale::get_locale());
-#endif
-        lm::setup_logging(os);
+        try {
+            setup_logging_internal(os);
+        } catch (...) {
+            for (auto var : lc_vars) {
+                environment::clear(var);
+            }
+            environment::set("LANG", "C");
+            setup_logging_internal(os);
```

Comment:
This will get caught by our normal "catch and log from main" mechanism which... actually can't log if we can't set this up. 

I think the right thing to do here is to raise a special exception that can be caught by main and cause an abort() (after writing to stderr in lieu of logging), OR by the ruby bindings to throw a LoadError or something like that on the `require facter` that brought us here.